src/Entity/BannerSelection/Campaign.php

```diff
@@ -35,7 +35,22 @@ public function getEnd(): \DateTime {
 		return $this->end;
 	}
 
-	public function isActiveAtPointInTime( \DateTime $time ): bool {
-		return $time->getTimestamp() > $this->start->getTimestamp() && $time->getTimestamp() < $this->end->getTimestamp();
+	public function getStart(): \DateTime {
```

Comment:
You should not expose state to the outside. Other than `getEnd`, this date is not used anywhere else and if some interaction with the start date is needed, it's better to have a meaningful method (like `isInActiveDateRange`) with a parameter than slurping out the data and then doing something with it. [Tell, don't ask](https://martinfowler.com/bliki/TellDontAsk.html)

getEnd` is a convenient shortcut, but also dangerous.